packages/fhir/src/Adaptor.js

```diff
@@ -172,6 +172,56 @@ export function getClaim(claimId, query, callback = s => s) {
     );
   };
 }
+
+/**
+ * A function that would extract all resources of the specified type from the bundle.
+ * @example
+ * extractResource({bundle},'resouceType')
+ * @param {object} bundle - Specify the bundle to extract resources from
+ * @param {string} resourceType - Specify the type of resource to be
+ * @param callback - (Optional) Handle the response from the server
+ * @return {Operation}
+ */
+export function extractResource(bundle, resourceType, callback) {
+  return state => {
+    const [resolvedBundle, resolveResourceType] = expandReferences(
+      state,
+      bundle,
+      resourceType
+    );
+
+    // Filter resources of the specified type
+    const extractedResources = (resolvedBundle.entry || []).filter(
+      entry =>
+        entry.resource && entry.resource.resourceType === resolveResourceType
+    );
+
+    const extractedBundle = {
+      resourceType: 'Bundle',
+      type: 'collection',
```

Comment:
This will support only resource type `collection` Can you modify this so that it can use other resource types @mahaomolise92

Reply:
Presumably we can just pass `resolvedResourceType` through?